File: Windows/OpenIoT.App/OpenIoT.Lib/Board/Protocol/Events/IPropertyTransmissionProtocolEvents.cs
using Palitri.OpenIoT.Board.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Palitri.OpenIoT.Board.Protocol.Events
{
    public interface IPropertyTransmissionProtocolEvents
    {
        void OnPingBack(object sender, byte[] data);
        void OnPropertyInfoReceived(object sender, BoardProperty p);
        void OnAllPropertiesInfoReceived(object sender);
        void OnSubscribedPropertyValueChanged(object sender, BoardProperty p, object oldValue);
        void OnInfoReceived(object sender, string info);
        void OnPropertiesSubscriptionSet(object sender);
        void OnPropertiesChangedSubscriptionReset(object sender);
    }
}

File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/board/protocol/events/IPropertyTransmissionProtocolEvents.java
package com.palitri.openiot.construction.framework.board.protocol.events;

import com.palitri.openiot.construction.framework.board.models.BoardProperty;

public interface IPropertyTransmissionProtocolEvents {
    void onPingRequest(Object sender, byte[] data);
    void onPingBack(Object sender, byte[] data);
    void onPropertyInfoReceived(Object sender, BoardProperty p);
    void onAllPropertiesInfoReceived(Object sender);
    void onSubscribedPropertyValueChanged(Object sender, BoardProperty p, Object oldValue);
    void onInfoReceived(Object sender, String info) ;
    void onPropertiesSubscriptionSet(Object sender);
    void onPropertiesChangedSubscriptionReset(Object sender);
}
